src/socks/__init__.py

```diff
@@ -51,46 +49,71 @@
 _defaultproxy = None
 _orgsocket = socket.socket
 
-class ProxyError(Exception): pass
-class GeneralProxyError(ProxyError): pass
-class Socks5AuthError(ProxyError): pass
-class Socks5Error(ProxyError): pass
-class Socks4Error(ProxyError): pass
-class HTTPError(ProxyError): pass
+
+class ProxyError(Exception):
+    """Base class for other ProxyErrors"""
+    pass
+
+
+class GeneralProxyError(ProxyError):
+    """Handle a general proxy error"""
+    pass
+
+
+class Socks5AuthError(ProxyError):
+    """Handle a SOCKS5 auth error"""
+    pass
+
+
+class Socks5Error(ProxyError):
+    """Handle a SOCKS5 non-auth error"""
+    pass
+
+
+class Socks4Error(ProxyError):
+    """Handle a SOCKS4 error"""
+    pass
+
+
+class HTTPError(ProxyError):
+    """Handle a HTTP error"""
+    pass
+
 
 _generalerrors = ("success",
-    "invalid data",
-    "not connected",
-    "not available",
-    "bad proxy type",
-    "bad input",
-    "timed out",
-    "network unreachable",
-    "connection refused",
-    "host unreachable")
+                  "invalid data",
+                  "not connected",
+                  "not available",
+                  "bad proxy type",
+                  "bad input",
+                  "timed out",
+                  "network unreachable",
+                  "connection refused",
+                  "host unreachable")
```

Comment:
```python
_generalerrors = (
    "success",
    "invalid data",
    ...
)
```

Reply:
I'm also a fan of hanging indentation, though our style is not consistent throughout the project. I've started creating a style guide for summarising our discussion for style ietms like this that are beyond pep8 but before manually changing one or two instances I would rather update the `fab code_quality` task to ensure the rule is enforced by autopep8 consistently.

If you want hanging indents to be the norm for autopep8 then we should (at a suitable time and after discussion) try updating setup.cfg and running code_quality:fix=1 on all affected files.

Reply:
The socks directory is mostly decommissioned, no need to look at its code quality. If someone could instead move the SOCKS DNS lookups to the asyncore thread, that should solve the issue. I did implement SOCKS DNS lookups in asyncore but never tested or integrated them. Perhaps it's not even necessary to support SOCKS DNS lookups, and instead keep the bootstrap DNS record in knownnodes, and update parts of the code which guess whether the record is an IPv4/IPv6/onion.